hips/hip-0019.md

```diff
@@ -0,0 +1,68 @@
+---
+hip: 0019
+title: "Enable Running Hooks in Parallel"
+authors: [ "Michael Morris <michael.morris@est.tech>" ]
+created: "2024-06-06"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Helm hooks are currently executed serially. In many cases there will be no reason why they cannot be run in parallel. Running hooks in parallel can significantly improve install/upgrade times
+
+## Motivation
+
+Decrease install/upgrade times by executing hooks in parallel
+
+## Specification
+
+Add a new flag to the install, upgrade, rollback and uninstall commands to enable parallel behavior for hook execution. The flag will be of integer type and will allow the user to specify the maximum number of hooks that can be executed in parallel.
```

Comment:
Consider [the different user roles for a Helm user](https://github.com/helm/community/blob/main/user-profiles.md). The person installing a chart may have little knowledge of how the package works. Installers of a chart aren't expected to know the business logic of the application. Instead, that knowledge is held by someone distributing the application.

Consider a classic package management situation like installing postgresql on a linux server. The one installing the package may not know where all the files need to live on the operating system, what configuration files need to go where, or the way this app should be added to startup as a daemon. Without having all that knowledge they are able to use the application.

Someone using the app may likely not know if hooks can be run in parallel. The one who would know that is the developer of the chart.

Can we bake this into the chart metadata so the responsibility is on the side of the chart developer instead of the chart user?

Reply:
Hi @mattfarina 
Proposed solution updated to address comment. Please take a look when you get time and let me know if there are any further comments